net/unbound/files/unbound.init

```diff
@@ -1,20 +1,718 @@
 #!/bin/sh /etc/rc.common
-# Copyright (C) 2016 Michael Hanselmann
+#
+##############################################################################
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+#   Copyright (C) 2016 Eric Luehrsen
+#
+##############################################################################
 
-START=61
+. /lib/functions/network.sh
 
+##############################################################################
+
+START=60
 USE_PROCD=1
+PROG="/usr/sbin/unbound"
+
+##############################################################################
+
+UNBOUND_B_CONTROL=0
+UNBOUND_B_DNSMASQ=0
+UNBOUND_B_DNSSEC=0
+UNBOUND_B_GATE_NAME=0
+UNBOUND_B_LOCL_BLCK=0
+UNBOUND_B_LOCL_NAME=0
+UNBOUND_B_LOCL_SERV=1
+UNBOUND_B_OBSCURE=0
+UNBOUND_B_PRIV_BLCK=1
+
+UNBOUND_D_RESOURCE=small
+UNBOUND_D_RECURSION=passive
+
+UNBOUND_TXT_FWD_ZONE=""
+UNBOUND_TTL_MIN=120
+
+UNBOUND_N_EDNS_SIZE=1280
+UNBOUND_N_FWD_PORTS=""
+UNBOUND_N_REC_PORT=53
+UNBOUND_N_ROOT_AGE=90
+
+UNBOUND_PID="/var/run/unbound.pid"
+UNBOUND_CONF="/var/etc/unbound.conf"
+UNBOUND_HEADER_CONF="/etc/unbound/header.conf"
+
+##############################################################################
+
+domain_ptr_ip6() {
```

Comment:
Its not magic or voodoo. The reason I started with the UCI update was to get dnsmasq DHCP/DNS as a sidecar to Unbound. That means forwarding name and address queries. So its a necessary component of the design purpose.

Built into busybox, I find awk is most efficient at text transpose, and thats all a PTR record is. I didn't want to add a package dependency on some bloated ip tool kit (text script is nothing in squashfs). 

If dnsmasq is serving interface "lan" at 192.168.0.1/24, 2001:db8:aa:3::1/64 and fd00:c0a8::1/64. Creating 0.168.192.in-addr.arpa is easy. Creating 0.0.0.0.0.0.0.0.8.a.0.c.0.0.d.f.ip6.arpa and 3.0.0.0.a.a.0.0.8.b.d.0.1.0.0.2.ip6.arpa robustly is PIA with the ip6 double colon flexibility. We need in-addr.arpa strings for "local-zone:" domain-insecure:" and "forward:" clauses.